creator-node/src/middlewares.js

```diff
@@ -266,18 +266,29 @@ async function ensureStorageMiddleware(req, res, next) {
 /**
  * Issue SyncRequests to both secondaries, and wait for at least one to sync before returning
  * @dev TODO - move out of middlewares layer
+ * @param ignoreWriteQuorum true if write quorum should not be enforced (don't fail the request if write quorum fails)
  */
-async function issueAndWaitForSecondarySyncRequests(req) {
+async function issueAndWaitForSecondarySyncRequests(
+  req,
+  ignoreWriteQuorum = false
+) {
   const serviceRegistry = req.app.get('serviceRegistry')
   const { snapbackSM } = serviceRegistry
 
   // Parse request headers
   const pollingDurationMs =
     req.header('Polling-Duration-ms') ||
     config.get('issueAndWaitForSecondarySyncRequestsPollingDurationMs')
+  // Write quorum header always takes precedence over env var if explicitly defined
+  // Empty/undefined header means enforceWriteQuorum decides if write quorum is enabled
+  const writeQuorumHeaderTrue = !!req.header('Enforce-Write-Quorum')
```

Comment:
maybe it's just me but i found this logic a bit difficult to follow. i would suggest much more explicit cases that match your comment above like

```
const writeQuorumHeaderTrue = !!req.header('Enforce-Write-Quorum')
const writeQuorumHeaderFalse = req.header('Enforce-Write-Quorum') === false
let enforceWriteQuorum = false

if (!ignoreWriteQuorum) {
  if (writeQuorumHeaderTrue) enforceWriteQuorum = true
  // !writeQuorumHeaderFalse is for undefined/null/empty values where it's not explicitly false
  else if (!writeQuorumHeaderFalse && config.get('enforceWriteQuorum') enforceWriteQuorum = true
}
```